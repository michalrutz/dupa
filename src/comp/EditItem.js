import React from "react";

const EditItem = ( {expense} ) => (
    <div>
        <p>{expense.description}</p>
        <p>{expense.amount}</p>
    </div>
)

export default EditItem;