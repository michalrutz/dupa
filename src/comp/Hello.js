import React from "react";

class Hello extends React.Component {
    render(){
        return (
            <div>
                <h1>Hello 4!</h1>
                {this.props.children}
            </div>
        )
    }
}
 

export default Hello;
