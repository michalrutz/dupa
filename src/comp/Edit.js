import React from "react";
import { connect } from "react-redux";
import EditItem from "./EditItem";

export const Edit = (props) => {
    return(
        <div>
            <EditItem key={props.expense.id} expense={props.expense}/>       
        </div> 
        )
}   
const mapStateToProps = (state, props) => ({
    expense:state.expense.find(
        (expense) => expense.id === props.match.params.id)
})

export default connect(mapStateToProps)(Edit);

// {props.expense.map(
//     (item) => {
//         if (item.id === props.match.params.id)
//         return <EditItem key={props.match.params.id} description={item.description} amount={item.amount}/>
//     } )}