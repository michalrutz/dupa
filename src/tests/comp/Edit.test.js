import React from "react";
import { Edit } from "../../comp/Edit";
import { shallow } from "enzyme";


let expenses = {description:"Tom", amount:40}

test("expenses with props", () => {
    const wrapper = shallow(<Edit expense={expenses}/>);
    expect(wrapper).toMatchSnapshot();
} )