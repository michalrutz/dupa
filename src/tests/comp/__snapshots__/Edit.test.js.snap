// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expenses with props 1`] = `
<div>
  <EditItem
    expense={
      Object {
        "amount": 0,
        "description": "none",
      }
    }
  />
</div>
`;
